qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -379,6 +379,13 @@ def populate_menus(self):
     def launch_custom_python_gui(self, filename):
         self.interface_executor.execute(open(filename).read(), filename)
 
+        # If Muon Analysis 2 window set parent manually
+        python_filename = filename.split('/')[-1]
+        if python_filename == "Muon_Analysis.py":
+            window = find_window("MuonAnalysis2", None)
+            window.setParent(self, window.windowFlags())
```

Comment:
This solves the issue of the interface dropping behind the main Mantid window on windows on workbench. It introduces the issue however that when minimized the window does not go into the system tray like would be expected instead you end up with a compressed window somewhere ontop of the main window. On windows this can be solved by removing the main window as a parent on minimization and then re-adding the main window as a parent on maximization. This was what was done for C++ interfaces on mantidplot. This won't work for operating systems that aren't windows however as they all seem to annoyingly handle windowing behavior differently. 

As most of the Muon users are on windows we may be able to get away with this but I think it is worth checking exactly what it does on the other operating systems. I know that when Martyn was working on a similar problem for plotting windows there was discussion about coming up with a generic way to handle this for interfaces as well. @martyngigg has anything come of that yet?

Reply:
I should say though that as far as I'm aware a nice solution without some compromises somewhere does not exist for the windows staying on top problem. I think the exact set of compromises decided on warrants discussion however.

Reply:
@Mantid-Matthew You are correct there are compromises with whatever we do but I would like us to be consistent with what being "on top" means and I would prefer not to have specific checks for filenames in the mainwindow like this.

I think in the end each interface will probably want to decide whether it should be on top or not. To be able to do this consistently then I think we will have to implement part or all of the original design for [interfaces in the workbench](https://github.com/mantidproject/documents/blob/master/Design/Workbench/design-classes.md) so that there is a single factory mechanism for all of them. #28493 is been put at the top of the next maintenance cycle and maybe we should look at how this can fit in with that too.

Reply:
Okay, for now I've reverted the change as I agree, it is a bit nasty/hacky having the check in the mainwindow. For now they can deal with the GUI window going behind workbench until a consistent solution exists.